src/main/groovy/org/transmartproject/batch/batchartifacts/AbstractSplittingItemReader.groovy

```diff
@@ -53,7 +54,9 @@ abstract class AbstractSplittingItemReader<T> extends ItemStreamSupport implemen
             while (!needsDelegateFetch()) {
                 def value = uncachedRead()
                 if (value != null) {
-                    cachedValues << value
+                    if(!earlyItemFilter || !earlyItemFilter.doFilter(value)) {
```

Comment:
ws and generally when we have a filter we keep the values for which the passed in passed function returns true, see for instance in [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter). It would be clear if you renamed `doFilter` to `keepItem` or `shouldKeepItem` , for instance.